src/xml/reader.cr

```diff
@@ -25,6 +25,14 @@ struct XML::Reader
     LibXML.xmlTextReaderRead(@reader) == 1
   end
 
+  def next
+    LibXML.xmlTextReaderNext(@reader) == 1
+  end
+
+  def next_sibling
```

Comment:
libXML docs say this and `next` are the same, are they? More docs in general would be great.

Reply:
No, they are not the same.

See this example:

```
<root>
  <childa/>
  <childb/><!-- current reader node -->
</root>
```

If you call `reader.next` would return `true` and it will move to `</root>`, but `reader.next_sibling` would return `false`, because there is no other sibling under the root node.

See:
* [xmlTextReaderNextSibling](https://github.com/GNOME/libxml2/blob/master/xmlreader.c#L2037)
* [xmlTextReaderNext](https://github.com/GNOME/libxml2/blob/master/xmlreader.c#L1669)

That's the theory at least, the system `/usr/lib/libxml2.2.dylib` on my mac which should be 2.9.4 according to `xml2-config --version` always returns `-1`, looks like I'll have to recompile crystal against the latest version from homebrew to see if it is a bug in that specifc version.

Reply:
Yeah, that appears to be a bug in libxml2, it works after applying the following patch to master:

```diff
diff --git a/xmlreader.c b/xmlreader.c
index 4053269b..c195d875 100644
--- a/xmlreader.c
+++ b/xmlreader.c
@@ -2037,7 +2037,7 @@ int
 xmlTextReaderNextSibling(xmlTextReaderPtr reader) {
     if (reader == NULL)
         return(-1);
-    if (reader->doc == NULL) {
+    if (reader->doc != NULL) {
         /* TODO */
        return(-1);
     }
```

I created GNOME/libxml2#13 to get this fixed in libxml.

Reply:
I have pushed a workaround that makes `next_sibling` work even with the incomplete `xmlTextReaderNextSibling()`.

**Note:** The libxml2 patch posted above is outdated and broken, see GNOME/libxml2#13 for the revised version.